docs/guides/federation.md

```diff
@@ -2,6 +2,175 @@
+to fetch all the books.
+
+Meanwhile `Book` used the `strawberry.federation.type` decorator, as opposed to
+the normal `strawberry.type`, this new decorator extend the base one and allows
+to define federation specific attributes to the type.
+
+In this case we are telling federation that the key to uniquely identify a book
+is the `id` field.
+
+> Federation keys can be thought primary keys. They are used by the gateway to
+> query types between multiple services and then joining them into the augmented
+> type.
+
+### Reviews service
+
+Let’s look at how our review service looks like, we want to define a type for a
+review but also extend the book to have a list of review.
+
+```python
+@strawberry.type
+class Review:
+    body: str
+
+def get_reviews() -> List[Review]:
+    return [Review(body="This is a review")]
+
+@strawberry.federation.type(extend=True, keys=["id"])
+class Book:
+    id: strawberry.ID = strawberry.federation.field(external=True)
+    reviews: List[Review] = strawberry.field(resolver=get_reviews)
+
+    @classmethod
+    def resolve_reference(cls, id: strawberry.ID):
+        return Book(id)
+```
+
+Now things are looking more interesting, the `Review` type is a GraphQL type
+that holds the content of the review.
+
+Meanwhile we are able to extend the `Book` type by using
+`strawberry.federation.type` again and passing `extend=True` as a parameter.
+This tells federation that we are extending an already existing type.
+
+We are also declaring two fields on `Book`, one is the `id` which is marked as
+external with `strawberry.federation.field(external=True)`, this tells
+federation that this field is not available in this service, and **that** it
+comes from another service.
+
+The other field is `reviews` which results in a list of `Reviews` for this book.
+
+Finally we also have a class method called `resolve_reference` that allows us to
+resolve references to types. The `resolve_reference` method is called when a
+GraphQL operation references an entities across multiple services. For example
+when doing this query:
+
+```graphql
+{
+  # query defined in the books service
+  books {
+    title
+    # field defined in the reviews service
+    reviews {
+      body
+    }
+  }
+}
+```
+
+The `resolve_reference` method is called with the `id` of the book for each book
+returned by the books service. The `id` is the field that has been defined as
+the key for the `Book` type. In our example we don't need to fetch additional
+data for the book so we create an instance of the Book type and return it.
+
+If we added more fields to book in a database this would be the place where we
+could query our database for the additional fields.
+
+The last thing we need to do is to define a `Query` type, even if our service
+only has one type that is not used directly in any GraphQL query. This is
+because any GraphQL server requires the Query type to be defined.
```

Comment:
Side note but I reckon it might be a good idea to remove this requirement because it feels kind of arbitrary and can be annoying when testing things like mutations. We can do it by removing the graphql-core schema validation rule.

Reply:
I think this might part of the spec :)

Reply:
Hmm you're right. That's a pain...